### PR TITLE
Feature/doc parser

### PR DESCRIPTION
**Pull request by JoshK2**
## Proposed Changes

- move jsDoc parser from React parser to a separate folder.
- Move the react parser and tests to a separate folder, so now the `parser.ts` file is only the entry file and call the React, Vue and jsDoc parser when is necessary. 
- Update `react-docgen` from 2.21.0 to 4.1.1  and fix Babel error related to this update.

These changes are for fixing this [issue](https://github.com/teambit/bit/issues/2105), have a better structure in the parser section and to be up to date with the `react-docgen` library so Bit can support more React features.

---
This change is [Reviewable](https://reviewable.io/reviews/teambit/bit/2118)

### CODE REVIEW

**Comment by JoshK2** [comment]
This pr fixing this issue: https://github.com/teambit/bit/issues/2105

File: src/jsdoc/parser.ts [hunk 1]
```diff
@@ -1,280 +1,17 @@
...
-  } catch (e) {
-    // never mind, ignore the doc of this source
-    logger.debug(`failed parsing docs using on path ${filePath} with error`, e);
+  const reactDocs: Doclet | [] = await reactParse(data, filePath);
+  if (Object.keys(reactDocs).length > 0) {
+    return reactDocs;
   }
-  // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
-  return doclets.filter(doclet => doclet.access === 'public');
+  return jsDoc(data, filePath);
```
Comment (GiladShoham) [inline src/jsdoc/parser.ts]: change this from jsDoc to jsDocParse

File: src/jsdoc/react/react-parser.ts [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    // common syntax `?` (e.g. `string?`)
+    formattedType = formattedType.replace('=', '?');
+  }
+  // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
+  tag.type = formattedType;
+
+  return tag;
+}
+
+function extractDataRegex(doc: string, doclets: Array<Doclet>, filePath: PathOsBased) {
```
Comment (GiladShoham) [inline src/jsdoc/react/react-parser.ts]: This function exist twice, here and in the jsdoc parse. consolidate them

File: src/jsdoc/parser.ts (hunk 1 above)
Comment (JoshK2) [reply src/jsdoc/parser.ts]: I fixed this

File: src/jsdoc/react/react-parser.ts (hunk 2 above)
Comment (JoshK2) [reply src/jsdoc/react/react-parser.ts]: I fixed this

**Pull request merged**